### ISSUE TITLE
Doctrine Migrations

### ISSUE THREAD

**Issue by cmodijk** [VasekPurchart/Doctrine-Date-Time-Immutable-Types#4]
I'm not sure why but every time we create a diff using doctrine migrations al the fields with `datetime_immutable` are being changed and updated. I don't know why this i'm hopping somebody know's how to fix this.

**Comment by VasekPurchart** [comment]
Hi, thanks for reporting this, before I look at it please tell me which command do you exactly mean, so that I can reproduce this. Do you mean using `app/console doctrine:migrations:diff`? Or `app/console doctrine:schema:update --dump-sql` ? If the former, can you please try the latter and tell me if it's also having the same issue?

**Comment by cmodijk** [comment]
Sorry, Yes both commands keep changing the database.

This is my entity

```
class Subscription
{
    /**
     * @var int
     *
     * @ORM\Column(name="pkSubscriptionId", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue
     */
    private $id;

    /**
     * @var Price
     *
     * @ORM\Embedded(class="DataBundle\Entity\ValueHolder\Price")
     */
    protected $price;

    /**
     * @var Periode
     *
     * @ORM\Embedded(class="DataBundle\Entity\ValueHolder\Periode")
     */
    protected $periode;

    /**
     * @var \DateTimeImmutable
     *
     * @ORM\Column(type="datetime_immutable", nullable=true)
     */
    private $canceledAt;

    /**
     * @var \DateTimeImmutable
     *
     * @ORM\Column(type="datetime_immutable", nullable=true)
     */
    private $startedAt;

    /**
     * @var \DateTimeImmutable
     *
     * @ORM\Column(type="datetime_immutable", nullable=true)
     */
    private $renewedAt;
```

Results of `app/console doctrine:schema:update --dump-sql`

```
ALTER TABLE subscriptions CHANGE canceledAt canceledAt DATETIME DEFAULT NULL, CHANGE startedAt startedAt DATETIME DEFAULT NULL, CHANGE renewedAt renewedAt DATETIME DEFAULT NULL;

```

**Comment by VasekPurchart** [comment]
Hey @cmodijk, I finally had time to look into this and I have probably found a suitable solution.

But first let me ask - why is your project not using [the bundle](https://github.com/VasekPurchart/Doctrine-Date-Time-Immutable-Types-Bundle) which was prepared to ease the integration? Is your project not using Symfony? I am mentioning this because I think this isssue would not occur, if you were using it. Am asking because this depends on the particular setup and bootstrapping process you have for Doctrine.

The core of the issue is that Doctrine has mapped multiple DBAL types for the same database type (datetime) and does not know which one is used. In order to know it it must save the used type in the schema - precisely in a SQL comment for that column. I have prepared a branch, where this is solved to be done automatically, could you please try it in your project, if it is sufficient to solve this issue? The branch is [sql-comments](https://github.com/VasekPurchart/Doctrine-Date-Time-Immutable-Types/tree/sql-comments) and it should be enough to just reference the branch in composer.json: `"vasek-purchart/doctrine-date-time-immutable-types": "dev-sql-comments"`. If you confirm that it's ok, I am going to release a new version ASAP, so you don't have to rely on this branch.

**Comment by cmodijk** [comment]
@VasekPurchart We are using the symfony 2 bundle but we can't use the replace method because FOSUserBundle uses an [instanceof check on \DateTime](https://github.com/FriendsOfSymfony/FOSUserBundle/blob/master/Model/User.php#L528) and if we use the replace method this wound not work any more. For our own entity's we are know using the `datetime_immutable` but there are other extensions like the [timestampable extension](https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/timestampable.md) which also rely on the \DateTime instance.

I wil try you branch tomorrow thx for the response!

**Comment by VasekPurchart** [comment]
In the bundle, you can [configure](https://github.com/VasekPurchart/Doctrine-Date-Time-Immutable-Types-Bundle#configuration) to just add the types, which is also the default option (exactly because of compatibility reasons). So give it a try! :) (but test the mentioned branch anyway, please).

**Comment by Erikvv** [comment]
I had this issue and I can confirm that your branch fixes it.
(I do not use symphony)

**Comment by VasekPurchart** [comment]
I released this fix in `0.1.1` thanks @cmodijk for reporing and @Erikvv for the verification.
